weave/trace_server/migrations/002_add_deleted_at.up.sql

```diff
@@ -0,0 +1,78 @@
+/*
+This migration adds the deleted_at column to: 
+    - the object_versions, call_parts, and calls_merged tables
+    - the object_versions_deduped and calls_merged_view views
+*/
+
+ALTER TABLE object_versions
+    ADD COLUMN deleted_at Nullable(DateTime64(3)) DEFAULT NULL;
+
+CREATE OR REPLACE VIEW object_versions_deduped as
+    SELECT project_id,
+        object_id,
+        created_at,
+        deleted_at,  -- **** Add deleted_at to the view ****
+        kind,
+        base_object_class,
+        refs,
+        val_dump,
+        digest,
+        if (kind = 'op', 1, 0) AS is_op,
+        row_number() OVER (
+            PARTITION BY project_id,
+            kind,
+            object_id
+            ORDER BY created_at ASC
+        ) AS _version_index_plus_1,
+        _version_index_plus_1 - 1 AS version_index,
+        count(*) OVER (PARTITION BY project_id, kind, object_id) as version_count,
+        if(_version_index_plus_1 = version_count, 1, 0) AS is_latest
+    FROM (
+            SELECT *,
+                row_number() OVER (
+                    PARTITION BY project_id,
+                    kind,
+                    object_id,
+                    digest
+                    ORDER BY created_at ASC
+                ) AS rn
+            FROM object_versions
+        )
+    WHERE rn = 1 WINDOW w AS (
+            PARTITION BY project_id,
+            kind,
+            object_id
+            ORDER BY created_at ASC
+        )
+    ORDER BY project_id,
+        kind,
+        object_id,
+        created_at;
+
+ALTER TABLE call_parts
+    ADD COLUMN deleted_at Nullable(DateTime64(3)) DEFAULT NULL;
+
+ALTER TABLE calls_merged
+    ADD COLUMN deleted_at SimpleAggregateFunction(any, Nullable(DateTime64(3)));
+
+ALTER TABLE calls_merged_view MODIFY QUERY
+    SELECT project_id,
+        id,
+        anySimpleState(wb_run_id) as wb_run_id,
+        anySimpleState(wb_user_id) as wb_user_id,
```

Comment:
Challenge: what happens if the user doing the deletion is not the user who created the call?

Reply:
ah, correct @tssweeney